app/src/main/java/com/example/androidtest/Calc.kt

```diff
@@ -0,0 +1,109 @@
+import kotlin.math.pow
+
+var ptrL = 0
+var ptrOp = 0
+var ptrVal = 0
+var arrayLex = arrayOfNulls<String>(200)
+var opStack = arrayOfNulls<String>(200)
+var valStack = arrayOfNulls<String>(200)
+
+fun priorityRank(o: String?): Int {
+    var rank = -1
+    when (o) {
+        "(" -> rank = 0
+        "+", "-" -> rank = 1
+        "*", "/" -> rank = 2
+        "^" -> rank = 3
+    }
+    return rank
+}
+
+fun execResult() {
+    val a1: Double
+    val a2: Double
+    var result: Double
+    val v2: String? = valStack[--ptrVal]
+    var v1: String? = valStack[--ptrVal]
+    val operation: String? = opStack[--ptrOp]
+    a1 = v1!!.toDouble()
+    a2 = v2!!.toDouble()
+
+    result = 0.0
+
+    when (operation) {
+        "+" -> result = a1 + a2
+        "-" -> result = a1 - a2
+        "*" -> result = a1 * a2
+        "/" -> result = a1 / a2
+        "^" -> result = a1.pow(a2)
+    }
+    v1 = result.toString()
```

Comment:
в данном фрагменте использую else ветку тк для присваивания должны быть обработаны все кейсы, поэтому лучшим решением будет enum
```suggestion
    v1 = when (operation) {
        "+" -> a1 + a2
        "-" -> a1 - a2
        "*" -> a1 * a2
        "/" -> a1 / a2
        "^" -> a1.pow(a2)
        else -> error("unsupported operation")
    }.toString()
```